src/huggingface_hub/hf_api.py

```diff
@@ -1654,7 +1794,22 @@ def list_datasets(
             if isinstance(filter, DatasetFilter):
                 params = self._unpack_dataset_filter(filter)
             else:
-                params.update({"filter": filter})
+                params = self._build_dataset_filter(
```

Comment:
Same comment as for `list_models` (would prefer to have everything in `list_datasets`).